testing/testing.go

```diff
@@ -88,8 +86,18 @@ func GenUpgrader(n *swarm.Swarm) *tptu.Upgrader {
 	}
 }
 
+type mSwarm struct {
+	*swarm.Swarm
+	ps peerstore.Peerstore
+}
+
+func (s *mSwarm) Close() error {
+	s.ps.Close()
+	return s.Swarm.Close()
+}
+
 // GenSwarm generates a new test swarm.
-func GenSwarm(t *testing.T, ctx context.Context, opts ...Option) *swarm.Swarm {
+func GenSwarm(t *testing.T, opts ...Option) network.Network {
```

Comment:
This change will likely affect other modules so it would be nice if we could avoid it (see my comment above about using `testing.Cleanup`.

Reply:
Oh cool, I wasn't aware of `testing.Cleanup`. That's exactly what I've been looking for!

Reply:
I think it's relatively new.